app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt

```diff
@@ -114,14 +116,23 @@ class BrowserFragment : BaseBrowserFragment(), UserInteractionHandler {
 
     override fun onStart() {
         super.onStart()
+        val context = requireContext()
+
         val toolbarSessionObserver = TrackingProtectionOverlay(
-            context = requireContext(),
-            settings = requireContext().settings()
+            context = context,
+            settings = context.settings()
         ) {
             browserToolbarView.view
         }
         getSessionById()?.register(toolbarSessionObserver, this, autoPause = true)
         updateEngineBottomMargin()
+        FirstTimePwaObserver(
```

Comment:
You could also add a check for `shouldShowFirstTimePwaFragment` so you don't need to add the observer after it succeeded the first time.